symbolic_pofk/pytorch/linear_plus.py

```diff
@@ -0,0 +1,288 @@
+        * (2 * k_batch ** 2 * 2998**2 / 5 / Om) ** 2
+        * tk_eh ** 2
+    )
+
+    return pk_batch
+
+
+def logF_fiducial(k_batch, theta_batch):
+    """
+    Compute the emulated logarithm of the ratio between the true linear power spectrum 
+    and the Eisenstein & Hu 1998 fit for LCDM modified from implementation in linear.py (Bartlett et al. 2023).
+
+    Args:
+        :k_batch (torch.Tensor): tensor containing the k values to evaluate P(k) at [h / Mpc] with shape (n_k,1)
+        :theta_batch (torch.Tensor): tensor containing the parameters with shape (batch_size, 9),
+            the 9 parameters are :
+                As: 10^9 times the amplitude of the primordial P(k)
+                Om: The z=0 total matter density parameter
+                Ob: The z=0 baryonic density parameter
+                h: Hubble constant, H0, divided by 100 km/s/Mpc
+                ns: Spectral tilt of primordial power spectrum
+                mnu: Sum of neutrino masses [eV / c^2]
+                w0: Time independent part of the dark energy EoS
+                wa: Time dependent part of the dark energy EoS
+                a: The scale factor to evaluate P(k) at
+
+    Returns:
+        :logF (torch.Tensor): The emulated logarithm of the ratio between the true linear power spectrum
+    """
+
+    logF = lcdm_logF_fiducial(k_batch, theta_batch[:, :5])
+
+    return logF
+
+
+def plin_plus_emulated(k, theta_batch):
+    """
+    Fiducial power spectrum given in Bartlett et al. 2023 for a batch of parameters.
+
+    Args:
+        :k (torch.Tensor): k values [h/Mpc] with shape (n_k)
+        :theta_batch (torch.Tensor): tensor containing the parameters with shape (batch_size, 9),
+            the 9 parameters are :
+                As: 10^9 times the amplitude of the primordial P(k)
+                Om: The z=0 total matter density parameter
+                Ob: The z=0 baryonic density parameter
+                h: Hubble constant, H0, divided by 100 km/s/Mpc
+                ns: Spectral tilt of primordial power spectrum
+                mnu: Sum of neutrino masses [eV / c^2]
+                w0: Time independent part of the dark energy EoS
+                wa: Time dependent part of the dark energy EoS
+                a: The scale factor to evaluate P(k) at
+
+    Returns:
+        :Pk (torch.Tensor): computed fiducial power spectrum for each k and theta in the batch
+    """
+
+    # add dim to k so that it can be broadcasted with theta_batch
+    k_batch = k.unsqueeze(1).to(device)
+
+    eh = get_eisensteinhu_nw(k_batch, theta_batch)
+    D = get_approximate_D(k_batch, theta_batch)
+    logF_value = logF_fiducial(k_batch, theta_batch)
+
+    F_value = torch.exp(logF_value)
+    R_value = growth_correction_R(theta_batch)
+    log10_S_value = log10_S(k_batch, theta_batch)
+    S_value = torch.pow(10, log10_S_value)
+
+    Pk = eh * D ** 2 * F_value * R_value * S_value
+
+    return Pk.T
```

Comment:
When running this code, I get the following warning:

```
UserWarning: The use of `x.T` on tensors of dimension other than 2 to reverse their shape is deprecated and it will throw an error in a future release. Consider `x.mT` to transpose batches of matrices or `x.permute(*torch.arange(x.ndim - 1, -1, -1))` to reverse the dimensions of a tensor. (Triggered internally at /Users/runner/work/pytorch/pytorch/pytorch/aten/src/ATen/native/TensorShape.cpp:3641.)
```

What is the behaviour you want? Please can you choose the appropriate one and change to it?